### ISSUE TITLE
Put an EOL at end of files

### ISSUE THREAD

**Issue by vlakoff** [bcit-ci/CodeIgniter#3476]
By removing the closing tags, https://github.com/bcit-ci/CodeIgniter/commit/a3ffbbb75ab9403941e4f810703313432b3993cc was actually an improvement.

Now, more than 6 years later, it's time to get rid of these blocks. Just remove these blocks and put an EOL. It's a no-brainer.

PRs have been so much polluted because of this.

<br>On a side note, the licence block at the top of the files can be cumbersome. I mainly think of the [default controller](https://github.com/bcit-ci/CodeIgniter/blob/develop/application/controllers/Welcome.php) and the [config files](https://github.com/bcit-ci/CodeIgniter/tree/develop/application/config). It would be great to remove them, if this is possible. @jim-parry

**Comment by vince83110** [comment]
This should be a request not an issue no ?

**Comment by narfbg** [comment]
Not against it, although it would look a bit like we've just trolled anybody who has ever submitted a PR.

**Comment by kakysha** [comment]
@narfbg .. some day you need to stop trolling people with that, really. GitHub editor is unusable now and it makes this not so necessary rule more useless.

**Comment by narfbg** [comment]
Did you not read my comment? I said I'm not against the proposal.

I don't appreciate you implying that I do troll people with it, I'm simply applying the same rule that was forced on me when I first started contributing.

**Comment by Razican** [comment]
I have created a PR for addressing it, if you finally decide to do it. About the license, with the @license sentence is covered, without the need for all the comment block.

**Comment by vlakoff** [comment]
I had prepared changes for this too, but locally... I'll get in touch a bit later.

**Comment by narfbg** [comment]
@jim-parry Would you give a +1/-1 on this?

I'd make the call alone, but I suspect you care a bit more about this one.

**Comment by jim-parry** [comment]
+1 Remove closing blocks

The closing block at the end of PHP files has always felt contrived to me.
I am in favor of removing it//.

**Comment by jim-parry** [comment]
The comment block at the beginning of source code files feels awkward too, but it serves a useful purpose according to the guidance I have been able to find ... it is part of asserting "collective copyright", on behalf of BCIT, because BCIT would be the body to defend any copyright infringement action, and keeping it there kept our lawyers happy, and that was part of the long and convoluted struggle/process to get the CodeIgniter adoption blessed by BCIT's lawyers. What a mouthful!

I personally feel that a @license tag would serve a similar purpose, but need some reference or guidance that would persuade our lawyers to relax.

Anyone?

**Comment by narfbg** [comment]
I'd actually prefer to keep the license comment block on top of files.

It has been pointed out that a single license.txt file in the package is also just fine, but the advantage comes in when/if somebody takes a file out of the complete package and use it independently. It shows the license in a clear, unambiguous way, but also where the file has been taken from, and gives the ability for authors of derived works to just add themselves as co-authors of the code, claim copyrights *on the modified code*, etc.
And really, users are not expected to actually look at our code on a daily basis, so it's not *that* annoying, and if they do - they'd use search anyway.

However, this goes only for the files under *system/*, and that probably excludes *system/languages/* (you can't really put copyright restrictions on language strings ...). For anything under *application/*, the huge comment blocks on top are indeed annoying and it's on top of either boilerplate code or configuration values that, similar to language strings, you can't put restrictions on.

I guess EllisLab put the copyright comment blocks in *application/* only to make a clear distinction that *application/* files are not under the OSL license. Now that it's all MIT ... we no longer need to educate anybody about that.

TL;DR: Keep the blocks in *system/*, remove them from *application/*.

I'm closing this issue now since it's not really about the comment blocks and I'm merging #3490.

**Comment by webvitaly** [comment]
Why not just add a license.txt file into the root directory with the text of license.
This approach is adopted by WordPress and all other free CMS-tools and frameworks.
Sorry, but I think that license text at the top of each file is pretty awkward.
Removing licence text from each file will reduce size of framework and developing will be much better with no need to scroll this license each time.

**Comment by Razican** [comment]
I think that with the license file in the root of the project and with a @license tag in the docs in each file should be fine.

**Comment by narfbg** [comment]
I *just* explained why.

**Comment by Razican** [comment]
Yes, I understood, I'm just saying that I think it's unnecesary.
